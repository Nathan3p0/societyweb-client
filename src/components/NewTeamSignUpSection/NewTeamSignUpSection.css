.newSignup__section {
    display: flex;
    justify-content: space-around;
    padding: 3rem;
    flex-wrap: wrap;
    background: #EDE4AD;
    color: #333333;
}

.newSignup__section h3 {
    font-size: 2rem;
    color: #333333;
}

.newSignup__section-benefits {
    padding-left: 2rem;
    max-width: 500px;
    width: 100%;
    align-self: flex-start;
    justify-self: center;
    margin-bottom: 1rem;
}

.newSignup__section-form {
    max-width: 500px;
    width: 100%;
    margin-bottom: 2rem;
}