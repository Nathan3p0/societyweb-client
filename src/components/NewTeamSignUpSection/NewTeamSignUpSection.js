import React, { Component } from 'react';
import NewTeamSignupForm from '../NewTeamSignupForm/NewTeamSignupForm';
import AuthApiService from '../../services/auth-api-service';
import './NewTeamSignUpSection.css'

class NewTeamSignUpSection extends Component {
    state = {
        error: null
    }

    handleTeamSignupSubmit = (e) => {
        e.preventDefault();
        this.setState({
            error: null,
            success: null
        });

        const { fullname, username, password, email, phone, group_name } = e.target;

        const newTeam = {
            full_name: fullname.value,
            username: username.value,
            password: password.value,
            email: email.value,
            phone: phone.value,
            group_name: group_name.value
        }

        AuthApiService.postNewTeam(newTeam)
            .then(res => {
                fullname.value = '';
                username.value = '';
                password.value = '';
                email.value = '';
                phone.value = '';
                group_name.value = '';
                this.setState({
                    success: true
                })
            })
            .catch(res => {
                this.setState({
                    error: res.error
                })
            })
    }

    render() {
        const { error, success } = this.state
        return (
            <section className="newSignup__section">
                <div className="newSignup__section-benefits">
                    <h3>The benefits of Society Web</h3>
                    <p>Lorem ipsum dolor sit amet consectetur, adipisicing elit. Error sed, explicabo molestiae iste consectetur mollitia voluptate voluptatum, est eius modi saepe, ad exercitationem numquam harum quo quasi asperiores repellat incidunt ab fuga sit id sapiente aspernatur similique! Nostrum enim quibusdam voluptatum ea velit odit impedit consequuntur totam adipisci, mollitia eos!</p>
                </div>
                <div className="newSignup__section-form">
                    <h3>New Team Signup</h3>
                    {error && <p className="error">{error}</p>}
                    {success && <p>Signup was successful. Click Leader Login at the top.</p>}
                    <NewTeamSignupForm handleSubmit={this.handleTeamSignupSubmit} />
                </div>
            </section>
        );
    }
}

export default NewTeamSignUpSection;