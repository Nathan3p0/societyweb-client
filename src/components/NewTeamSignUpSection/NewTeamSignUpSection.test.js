import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import NewTeamSignUpSection from './NewTeamSignUpSection'

describe('Member Login Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<NewTeamSignUpSection />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<NewTeamSignUpSection />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})