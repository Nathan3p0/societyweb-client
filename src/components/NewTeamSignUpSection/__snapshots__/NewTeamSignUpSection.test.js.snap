// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Team Signup Section Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="newSignup__section"
>
  <div
    className="newSignup__section-benefits"
  >
    <h3>
      The benefits of Society Web
    </h3>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Error sed, explicabo molestiae iste consectetur mollitia voluptate voluptatum, est eius modi saepe, ad exercitationem numquam harum quo quasi asperiores repellat incidunt ab fuga sit id sapiente aspernatur similique! Nostrum enim quibusdam voluptatum ea velit odit impedit consequuntur totam adipisci, mollitia eos!
    </p>
  </div>
  <div
    className="newSignup__section-form"
  >
    <h3>
      New Team Signup
    </h3>
    <form
      onSubmit={[Function]}
    >
      <ul
        className="signup__form"
      >
        <li>
          <label
            htmlFor="fullname"
          >
            Full Name:
          </label>
          <input
            id="fullname"
            name="full_name"
            placeholder="name"
            required={true}
            type="text"
          />
        </li>
        <li>
          <label
            htmlFor="username"
          >
            Username:
          </label>
          <input
            id="username"
            name="username"
            placeholder="username"
            required={true}
            type="text"
          />
        </li>
        <li>
          <label
            htmlFor="password"
          >
            Password:
          </label>
          <input
            id="password"
            name="password"
            placeholder="password"
            required={true}
            type="password"
          />
        </li>
        <li>
          <label
            htmlFor="email"
          >
            Email:
          </label>
          <input
            id="email"
            name="email"
            placeholder="email"
            required={true}
            type="text"
          />
        </li>
        <li>
          <label
            htmlFor="phone"
          >
            Phone Number:
          </label>
          <input
            id="phone"
            name="phone"
            placeholder="123-455-0505"
            required={true}
            type="tel"
          />
        </li>
        <li>
          <label
            htmlFor="group_name"
          >
            Team Name
          </label>
          <input
            id="group_name"
            name="group_name"
            placeholder="team name"
            required={true}
            type="text"
          />
        </li>
        <li>
          <button
            type="submit"
          >
            Sign Up
          </button>
        </li>
      </ul>
    </form>
  </div>
</section>
`;
